lib/spack/llnl/util/filesystem.py

```diff
@@ -2236,30 +2236,27 @@ def link_dest(self):
         if hasattr(self.pkg, "libs") and self.pkg.libs:
             pkg_libs = set(self.pkg.libs.directories)
         else:
-            pkg_libs = set((self.pkg.prefix.lib, self.pkg.prefix.lib64))
+            pkg_libs = set((self.pkg.prefix.lib, self.pkg.prefix.lib64, self.pkg.prefix.bin))
+        return pkg_libs
 
-        return pkg_libs | set([self.pkg.prefix.bin]) | self.internal_links
-
-    @property
-    def internal_links(self):
-        """
-        linking that would need to be established within the package itself. Useful for links
-        against extension modules/build time executables/internal linkage
-        """
-        return self._internal_links
-
-    def add_internal_links(self, *dest):
+    def add_library_dependent(self, *dest):
         """
-        Incorporate additional paths into the rpath (sym)linking scheme.
-
-        Paths provided to this method are linked against by a package's libraries
-        and libraries found at these paths are linked against a package's binaries.
-        (i.e. /site-packages -> /bin and /bin -> /site-packages)
+        Add paths to set of common paths that need to link against other libraries
 
-        Specified paths should be outside of a package's lib, lib64, and bin
+        Specified paths should be outside of a package's common link paths, i.e. the lib, lib64, and bin
         directories.
         """
```

Comment:
Can `dest` include paths that aren't directories? I assume we could accept either and then extract a directory for paths that are not directories, but I think it's worth mentioning either way in the docstring.